pkg/file/joiner/internal/job.go

```diff
@@ -110,12 +107,24 @@ func (j *SimpleJoinerJob) nextReference(level int) error {
 	chunkAddress := swarm.NewAddress(data[cursor : cursor+swarm.SectionSize])
 	err := j.nextChunk(level-1, chunkAddress)
 	if err != nil {
-		return err
+		// if the last write is a "dangling chunk" the data chunk will have been moved
+		// to an intermediate level. In this edge case, the error must be suppressed,
+		// and the cursor manually to data length boundary to terminate the loop in
+		// the calling frame.
+		if j.readCount+int64(len(data)) == j.spanLength {
+			j.cursors[level] = len(j.data[level])
+			//j.dataC <- data
+			err = j.sendChunkToReader(data)
```

Comment:
`return j.sendChunk...`?

Reply:
I would like to log the error if it's not EOF